src/freva/_esgf.py

```diff
@@ -12,102 +12,356 @@
+    for each dataset.
+
+    Parameters:
+    -----------
+    show_facet: Union[str, list[str]], default: None
+        List all values for the given facet (might be
+        defined multiple times). The results show the possible
+        values of the selected facet according to the given
+        constraints and the number of *datasets* (not files)
+        that selecting such value as a constraint will result
+        (faceted search)
+    **search_constraints: Union[str, Path, in, list[str]]
+        Search facets to be applied in the data search.
+
+    Returns:
+    --------
+    dict[str, list[str]]:
+        Collection of facets
+
+    Example
+    -------
+
+    List the values of the attributes ``variable``
+    and ``time_frequency`` stored at the (DKRZ) local node
+    (``distrib=false``) holding the latest version (``latest=true``)
+    for a particular realization within the project ``CMIP6``:
+
+    .. execute_code::
+
+        import freva
+        facets = freva.esgf_facets(
+            mip_era="CMIP6",
+            activity_id="ScenarioMIP",
+            source_id="CNRM-CM6-1",
+            institution_id="CNRM-CERFACS",
+            experiment_id="ssp585",
+            distrib=False,
+            latest=True,
+            show_facet=["variable", "frequency"])
+        print(facets)
+
+    """
+    show_facet = show_facet or []
+    p2p = P2P()
+    return p2p.get_facets(show_facet, **search_constraints)
+
+
+def esgf_datasets(
+    **search_constraints: dict[str, str],
+) -> list[tuple[str, str]]:
+    """List the name of the datasets (and version) in the ESGF.
+
+    The method queries the ESGF nodes for dataset information.
+    The key=value syntax follows that of ``freva.databrowser``
+    but the key names follow the ESGF standards for each dataset.
+
+    Parameters:
+    -----------
+    **search_constraints: Union[str, Path, in, list[str]]
+        Search facets to be applied in the data search.
+
+    Returns:
+    --------
+    list[tuple[str, str]]:
+        list of (dataset_name, version_number) tuples
+
+    Example
+    -------
+
+    List the datasets corresponding to a query of files
+    stored at the (DKRZ) local node (``distrib=false``)
+    holding the latest version (``latest=true``) for a particular
+    realization within the project ``CMIP6``:
+
+    .. execute_code::
+
+        import freva
+        datasets = freva.esgf_datasets(
+            mip_era="CMIP6",
+            activity_id="ScenarioMIP",
+            source_id="CNRM-CM6-1",
+            institution_id="CNRM-CERFACS",
+            experiment_id="ssp585",
+            frequency="3hr",
+            variable="uas",
+            variant_label="r1i1p1f2",
+            distrib=False,
+            latest=True,
+            )
+        print(f"{len(datasets) =}")
+        for dataset in datasets[:5]:
+            print(dataset)
+    """
+    p2p = P2P()
+    return sorted(p2p.get_datasets_names(**search_constraints))
+
+
```

Comment:
Can you make that a handled exception by adding the `handled_exception` decorator like in the other functions?

Reply:
I will check how to do that now